test/test_examples.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+# Copyright (C) 2024 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+set -e
+
+WORKSPACE=$1
+BUILD_DIR=$2
+INSTALL_DIR=$3
+
+echo "Running: $0 $*"
+
+function print_usage() {
+	echo "$(basename $0) - test all examples standalone"
+	echo "Usage: $(basename $0) <workspace_dir> <build_dir> <install_dir>"
+}
+
+if [ "$3" == "" ]; then
+	echo -e "error: too few arguments\n"
+	print_usage
+	exit 1
+fi
+
+if [ ! -f $WORKSPACE/README.md ]; then
+	echo -e "error: incorrect <workspace_dir>: $WORKSPACE\n"
+	print_usage
+	exit 1
+fi
+
+WORKSPACE=$(realpath $WORKSPACE)
+BUILD_DIR=$(realpath $BUILD_DIR)
+INSTALL_DIR=$(realpath $INSTALL_DIR)
+
+shift 3
+EXAMPLES="$*"
+if [ "$EXAMPLES" == "" ]; then
+	echo "No examples to run!"
+	exit 0
+fi
+
+echo "Examples to run: $EXAMPLES"
+echo
+
+cd ${BUILD_DIR}
+echo "DIR=$(pwd)"
+
+set -x
+cmake .. -DCMAKE_INSTALL_PREFIX="$INSTALL_DIR"
```

Comment:
hmm this might be weird - we run cmake in our original build_dir, while executing ctest...?

Reply:
Done

Reply:
I still don't like executing "cmake" within a build_dir, while we're in the middle of execution of ctest, but it seems to work. We still have to re-write this for other OSes, so this bash script will be removed one day...

Reply:
Please review the fix: https://github.com/oneapi-src/unified-memory-framework/pull/611